gpflow/utilities/ops.py

```diff
@@ -102,6 +102,17 @@ def square_distance(X, X2):
     return dist
 
 
+def distance(X, X2):
+    """
+    Returns (X - X2ᵀ)
+
+    This function can deal with leading dimensions in X and X2.
```

Comment:
Could you add what the shape of the return value is, given shapes of the arguments? (Ideally asserted in code using `tf.debugging.assert_shapes` - see #1219 for some examples!)